lib/plugins/aws/package/compile/functions/index.test.js

```diff
@@ -2525,6 +2525,94 @@ describe('AwsCompileFunctions #2', () => {
     });
   });
 
+  describe('IamConfig object tests', async () => {
```

Comment:
Concerning tests, as indicated in main issue description, let's not put any effort in updating them in current shape.

Technically we're just refactoring internals, and result CloudFormation template should remain same (at least it should produce very same effect)

To avoid tests breaking on internal changes, we need to refactor affected tests to `runServerless` variant.

I've outlined on how to do that for two tests files that you're trying to update here Check: #8523 and #8524 

Until those are addressed, we need assume that this PR cannot be merged (but it doesn't stop us from finalizing this refactor - it just cannot be properly validated until tests are refactored).

After we refactor all affected tests, the only test changes that may be wanted in scope of this PR is moving some of the tests to other files, or improving the coverage, that's it, but I would decide that once we have all tests refactored, and functionality in this PR ready